libsrc/osic1p/cputc.s

```diff
@@ -36,10 +41,13 @@ L1:     cmp     #$0D            ; LF?
 cputdirect:
```

Comment:
Just to make sure there's no misunderstanding: When we discussed the usage of 'extra' modules to support different screen sizes I presumed that you'd create modules which export `cputdirect` in a different way. If you can however come up with a reasonable way to have only one implementeation of `cputdirect` just with different parameters coming form 'extra' modules than this is in general fine with me.

But please keep in mind that cc65 C library programming is mostly about size minimizing. If having several `cputdirect` implementations for diffrent screen sizes would result in (significant) smaller code size than the current approach then I'd argue to modify it. If you don't want to repeat similiar code paths then maybe phrasing the actual code as macro (which is called with different parameters) in the different modules might be a way to go. The various Atari TGI drivers follow a similiar pattern: https://github.com/cc65/cc65/tree/master/libsrc/atari/tgi

And finally I'd like to point out that CONIO is in general not required to support scrolling. From the API perspective the behaviour of writing beyond the lower right corner of the screen is just undefined. Programs doing it might simply crash or end up to continue writing in the upper left corner or <...>. The reason for that design decision is that the CONIO implementation is supposed to be small (and fast).

I don't know of the target(s) at hand but usually cc65 targets with text i/o capabilities come with some firmware allowing for (sometimes rather slow) character output - and that firmware code usually supports scrolling. So the usual approach is to map the firmware functionality (with scrolling) to POSIX I/O on stdout (and stderr) aka file number 1 (and 2)  and implement CONIO with as-fast-as-possible direct video RAM access (without scrolling).

In case you want to argue that the POSIX I/O approach doesn't apply to the target(s) at hand because of the missing mass storage support you should know that it is perfectly acceptable to only support stdin, stdout and stderr and have no open() / close() at all.